docs/spire.md

```diff
@@ -0,0 +1,273 @@
+ ▲           │     ┌───────┴───────────────────────────────┐     ▲
+ │           │     │           Tekton TaskRun              │     │
+ │           │     └───────────────────────────────────────┘     │
+ │  Configure│                                          ▲        │ Attest
+ │  Pod &    │                                          │        │   +
+ │  check    │                                          │        │ Request
+ │  ready    │     ┌───────────┐                        │        │ SVIDs
+ │           └────►│  TaskRun  ├────────────────────────┘        │
+ │                 │  Pod      │                                 │
+ │                 └───────────┘     TaskRun Entrypointer        │
+ │                   ▲               Sign Result and update      │
+ │ Get               │ Get SVID      TaskRun status with         │
+ │ SPIRE             │               signature + cert            │
+ │ server            │                                           │
+ │ Credentials       │                                           ▼
+┌┴───────────────────┴─────────────────────────────────────────────────────┐
+│                                                                          │
+│   SPIRE Agent    ( Runs as   )                                           │
+│   + CSI Driver   ( Daemonset )                                           │
+│                                                                          │
+└──────────────────────────────────────────────────────────────────────────┘
+```
+
+Initial Setup:
+1. As part of the SPIRE deployment, the SPIRE server attests the agents running on each node in the cluster.
+1. The Tekton Controller is configured to have workload identity entry creation permissions to the SPIRE server.
+1. As part of the Tekton Controller operations, the Tekton Controller will retrieve an identity that it can use to talk to the SPIRE server to register TaskRun workloads.
+
+When a TaskRun is created:
+1. The Tekton Controller creates a TaskRun pod and its associated resources
+1. When the TaskRun pod is ready, the Tekton Controller registers an identity with the information of the pod to the SPIRE server. This will tell the SPIRE server the identity of the TaskRun to use as well as how to attest the workload/pod.
+1. After the TaskRun steps complete, as part of the entrypointer code, it requests an SVID from SPIFFE workload API (via the SPIRE agent socket)
+1. The SPIRE agent will attest the workload and request an SVID.
+1. The entrypointer receives an x509 SVID, containing the x509 certificate and associated private key. 
+1. The entrypointer signs the results of the TaskRun and emits the signatures and x509 certificate to the TaskRun results for later verification.
+
+## Enabling TaskRun result attestations
+
+To enable TaskRun attestations:
+1. Make sure `enable-spire` is set to `"true"` in the `feature-flags` configmap, see [`install.md`](./install.md#customizing-the-pipelines-controller-behavior) for details
```

Comment:
Drive-by requests re. this option:
1. The name of this option and the values don't match the name in the TEP (https://github.com/tektoncd/community/blob/main/teps/0089-nonfalsifiable-provenance-support.md#enabling-spire-on-the-entrypointer-image) - i.e. this should be `--enforce-nonfalsifiablity=spire` - and in general please audit this functionality against TEP-0089 which describes the design that was proposed and accepted
2. Please also document new config options in https://github.com/tektoncd/pipeline/blob/main/docs/install.md

Reply:
> Drive-by requests re. this option:
> 
> 1. The name of this option and the values don't match the name in the TEP (https://github.com/tektoncd/community/blob/main/teps/0089-nonfalsifiable-provenance-support.md#enabling-spire-on-the-entrypointer-image) - i.e. this should be `--enforce-nonfalsifiablity=spire` - and in general please audit this functionality against TEP-0089 which describes the design that was proposed and accepted
Agree.. Modified the user facing flag enable-spire to enforce-nonfalsifiability.

> 2. Please also document new config options in https://github.com/tektoncd/pipeline/blob/main/docs/install.md

Reply:
Regarding the install.md, this feature is not yet done. We have 2 more PRs (one in pipelines and the other in chains) before this can be done.
The install.md is a user facing doc. I think we should do this around in the final PR. WDYT?